internal/testutil/server.go

```diff
@@ -17,11 +17,13 @@ func (s *Server) KSMEndpoint() string {
 	// We must add /metrics to the URL here as the KSM override endpoint must be a full URL
 	return s.Server.URL + "/ksm/metrics"
 }
+
 func (s *Server) KubeletEndpoint() string {
 	return s.Server.URL + "/kubelet"
 }
+
 func (s *Server) ControlPlaneEndpoint(component string) string {
-	return s.Server.URL + "/controlplane/" + strings.ReplaceAll(component, "/", "")
+	return s.Server.URL + path.Join("/controlplane", component, "metrics")
```

Comment:
why keeping the `s.Server.URL + path.Join` and not everything in the join?

Reply:
doesn't work for url just for path https://go.dev/play/p/qX93BzDNKXY